lib/Doctrine/ORM/Internal/Hydration/Cache/LazyPropertyMap.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Internal\Hydration\Cache;
+
+/**
+ * Concept taken from ocramius/lazy-map
+ *
+ * @link https://github.com/Ocramius/LazyMap/blob/1.0.0/src/LazyMap/CallbackLazyMap.php
+ * @internal do not use: internal class only
+ */
+final class LazyPropertyMap
```

Comment:
I should probably import tests for this one.

Reply:
Definitely 😉

Reply:
Done that.

Reply:
A benchmark for this class is still needed